### DISCUSSION TITLE
Custom Suggestions in search box from the given SharePoint list

### DISCUSSION THREAD

**Discussion by sayuri8** [microsoft-search/pnp-modern-search#2740] [Q&A]
Hi everyone,

I tried to use the pnp modern search box and search result. In the search box I have to type the exactly word to get the items from the SharePoint list otherwise it fails. Also it doesn't make any suggestions which starts with any letter then it should show the available items in the list. 

Can you maybe show how it works?

**Comment by wobba** [comment]
For suggestions you need a custom suggestion provider: https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_suggestions_provider/ If you want to add some wildcard support use this query template: `({searchTerms}) OR ({searchTerms}*)`

**Reply by sayuri8** [reply]
Thanks for your reply and where exactly do I have to upload this code? I don't see the custom suggestion provider as an option in search result. Is it normal that the results shows everything what it has, is it possible to show the result only if it's searched?

**Reply by wobba** [reply]
Good point as it's not well documented :) You deploy the sppkg in an app catalog, then reference the manifest id in page 4 of the search box web part property pane. This will make it appear on page 2 of the properties.

**Reply by sayuri8** [reply]
Thanks for your reply. You mean by app catalog the power app page?

**Reply by wobba** [reply]
https://learn.microsoft.com/en-us/sharepoint/use-app-catalog
